aion/lib/aion/room_channel/room.ex

```diff
@@ -76,12 +89,24 @@ defmodule Aion.RoomChannel.Room do
     room.question
   end
 
-  @spec get_new_question_with_answers(integer) :: %{question: Question.t, answers: list Answer.t}
-  defp get_new_question_with_answers(room_id) do
-      question = Question.get_random_question(room_id)
-      answers = Answer.get_answers(question.id)
-      Logger.debug fn -> "Answers: #{inspect(Enum.map(answers, fn answer -> answer.content end))}" end
+  @spec get_new_question_with_answers(question_list, integer) ::  new_question_set
+  defp get_new_question_with_answers(questions, room_id) do
+    [new_question | remaining_questions] = questions
+    answers = Answer.get_answers(new_question.id)
+    Logger.debug fn -> "Answers: #{inspect(Enum.map(answers, fn answer -> answer.content end))}" end
+
+    remaining_questions = if Enum.empty?(remaining_questions), do: fetch_questions(room_id), else: remaining_questions
+
+    %{questions: remaining_questions, question: new_question, answers: answers}
+  end
+
+  @spec get_new_questions(integer) :: %{questions: question_list}
+  defp get_new_questions(room_id) do
+    %{questions: Enum.shuffle(fetch_questions(room_id))}
+  end
 
-      %{question: question, answers: answers}
+  @spec fetch_questions(integer) :: question_list
+  defp fetch_questions(room_id) do
+    Question.get_questions_by_room_id(room_id)
```

Comment:
This function is not introducing any additional logic. In this case you could try using [defdelegate](https://hexdocs.pm/elixir/Kernel.html#defdelegate/2). 

The decision is up to you - if you think the get_questions_by_room_id is not a suitable name here you could try making a defdelegate and this way provide a function named fetch_questions (which will basically work as an alias for the external function). If you think it's not needed, you can just replace fetch_questions with Question.get_questions_by_room_id.